### ISSUE TITLE
Strs clashes with Compat

### ISSUE THREAD

**Issue by bkamins** [JuliaString/Strs.jl#95]
As follows:
```
julia> using Strs

julia> using Compat
WARNING: Method definition in(Any) in module Strs overwritten in module Compat
WARNING: Method definition ==(Any) in module Strs overwritten in module Compat
WARNING: Method definition contains(AbstractString, Base.Regex) in module Strs overwritten in module Compat
WARNING: Method definition isequal(Any) in module Strsoverwritten in module Compat
```

**Comment by ScottPJones** [comment]
Not sure what I can do about that - I've been running into problems with some overbroad definitions in Compat, and Compat does not check to make sure it isn't overwriting some other definition.

**Comment by bkamins** [comment]
But I guess the reason is that you are doing "type piracy" thing here that *Compat.jl* assumes does not happen. E.g. you define a general `isequal(x)` function that acts on any argument `x`. Or I am missing something?

**Comment by ScottPJones** [comment]
The definitions are in my own "compat.jl" file, which does the same as Compat.jl.
I suppose I can make it use 'Compat.jl' on v0.6.2, but I'd wanted to avoid having any dependencies on other packages (and often Compat.jl takes a while before fixing problems, like you may have seen with the changes to `round` and `signif` that cause problems for other packages in JuliaStrings).

**Comment by bkamins** [comment]
Up to your decision. Probably in the long run it is cleaner to use Compat.jl.

**Comment by ScottPJones** [comment]
This will all go away, as not long after v0.7 is released, I'll probably drop v0.6.2 support.
